### DISCUSSION TITLE
Does updating the binary of a running service like NGINX via homebrew in turn force homebrew to stop or remove the old binary?

### DISCUSSION THREAD

**Discussion by paulzakin** [Homebrew/discussions#3362] [Everyday usage] [answered]
For example, let’s say that you’ve installed NGINX 1.2.4 via homebrew and you’ve just run `brew upgrade nginx`, installing NGINX 1.2.5. Does that stop the currently running version of NGINX at any point on your system, or is it required to run something like `brew services restart nginx` first? I ask mostly because I know homebrew doesn’t keep older versions of packages around for very long, and I’m unclear how that interacts when an older version of the package is running.

**Comment by SMillerDev** [comment, marked as answer]
For running services homebrew will print a message urging you to restart the service. Restarts are not automated.

If the cleanup decides to remove the old binary that is being ran it'll most likely crash. Some services are set to restart after a crash and that might make it functional again, but I wouldn't depend on it.

**Reply by paulzakin** [reply]
Got it, that makes sense. Thanks!
